### PR TITLE
chore(ci): automate tpch benchmark with cloud

### PR DESCRIPTION
**Pull request by huangjw806**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As the title.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: ci/scripts/benchmark.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+trap cleanup EXIT
+
+if [[ -z "${RISINGWAVE_IMAGE_TAG+x}" ]]; then
+  IMAGE_TAG="latest"
+else
+  IMAGE_TAG="${RISINGWAVE_IMAGE_TAG}"
+fi
+
+date=$(date '+%Y%m%d-%H%M%S')
+TENANT_NAME="bench-${date}"
```
Comment (sumittal) [inline ci/scripts/benchmark.sh:55]: are we going to use the same format of tenant name for single and multinode benchmark runs?

File: ci/scripts/benchmark.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+echo "--- RWC Create a Risingwave Instance"
+rwc t create -tenant ${TENANT_NAME} -sku ${SKU} -imagetag ${IMAGE_TAG}
+
+sleep 2
+
+echo "--- Wait Risingwave Instance Ready "
+endpoint=$(rwc t get-endpoint -tenant ${TENANT_NAME})
+polling ${endpoint}
+
+echo "--- Generate Tpch-Bech Args"
```
Comment (BugenZhao) [inline ci/scripts/benchmark.sh:80]:
```suggestion
echo "--- Generate Tpch-Bench Args"
```

File: ci/docker-compose.yml [hunk 3]
```diff
@@ -14,6 +14,33 @@ services:
...
+  zookeeper:
+    image: confluentinc/cp-zookeeper
+    ports:
+      - "2181:2181"
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 2181
+      ZOOKEEPER_TICK_TIME: 2000
+      ZOOKEEPER_SYNC_LIMIT: 2
+
+  kafka:
```
Comment (skyzh) [inline ci/docker-compose.yml:26]: Running Kafka here might lead to high network fee. Better to use a hosted Kafka service from AWS.

Comment (skyzh) [reply ci/docker-compose.yml:26]: Just ensure Kafka is running in the same region as RisingWave Cloud, then everything should be fine.

Comment (huangjw806) [reply ci/docker-compose.yml:26]: All services and processes of the automated benchmark are in the same region, so there should be no excessive network charges.

**Review by skyzh — APPROVED** [review]
LGTM. May merge first and see if it works. There could be many problems with this approach, e.g. how can RisingWave cloud connects to Kafka in Buildkite's VPC, vice versa? We need to test after get this PR merged.

Also I'm wondering whether it's necessary to host the benchmark workflows in RisingWave main repo. Should it goes into a private / separate repo instead?

**Comment by huangjw806** [comment]
> LGTM. May merge first and see if it works. There could be many problems with this approach, e.g. how can RisingWave cloud connects to Kafka in Buildkite's VPC, vice versa? We need to test after get this PR merged.
> 
> Also I'm wondering whether it's necessary to host the benchmark workflows in RisingWave main repo. Should it goes into a private / separate repo instead?

I have tested on this branch and it works. Also, I don't think it's necessary to put it in the RisingWave main repo, maybe we'll move it to the risingwave-test repo later.

**Review by sumittal — COMMENTED** [review]
Looks good!

**Pull request merged**